### PR TITLE
[Distributed] More docs for ad-hoc requirements

### PR DESCRIPTION
**Pull request by ktoso**
Missed some spots for documenting more of the ad-hoc requirements, adding now.

rdar://99899213

### CODE REVIEW

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
@swift-ci please smoke test

**Comment by ktoso** [comment]
resolves rdar://99899213

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 1]
```diff
@@ -688,6 +702,32 @@ func _executeDistributedTarget<D: DistributedTargetInvocationDecoder>(
 /// Once encoded, the system should use some underlying transport mechanism to send the
 /// bytes serialized by the invocation to the remote peer.
 ///
+/// ### Protocol requirements
+/// Similar to the ``DistributedActorSystem`` and its `remoteCall` and `remoteCallVoid` protocol requirements,
+/// the `DistributedTargetInvocationEncoder` contains a few methods which are not possible to express in source due to
+/// advanced use generics combined with associated types. Specifically, the `recordArgument` and `recordReturnType`
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:708]: Nit: use of

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 2]
```diff
@@ -688,6 +702,32 @@ func _executeDistributedTarget<D: DistributedTargetInvocationDecoder>(
...
+/// methods are not expressed in source as protocol requirements, but will be treated by the compiler as-if they were.
+///
+/// > Note: Although the `recordArgument` method is not expressed as protocol requirement in source,
+/// > the compiler will provide the same errors as-if they were declared explicitly in this protocol.
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:712]: `it was`

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 3]
```diff
@@ -771,10 +820,48 @@ public struct RemoteCallArgument<Value> {
 
 /// Decoder that must be provided to `executeDistributedTarget` and is used
 /// by the Swift runtime to decode arguments of the invocation.
+///
+/// ### Protocol requirements
+/// Similar to the ``DistributedTargetInvocationEncoder`` and its `recordArgument` and `recordReturnType` protocol requirements,
+/// the `DistributedTargetInvocationDecoder` contains a method which is not possible to express in source due to
+/// advanced use generics combined with associated types. Specifically, the `decodeNextArgument`
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:827]: Nit: use of

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 4]
```diff
@@ -771,10 +820,48 @@ public struct RemoteCallArgument<Value> {
...
+/// method is not expressed in source as protocol requirement, but will be treated by the compiler as-if it was.
+///
+/// > Note: Although the `decodeNextArgument` method is not expressed as protocol requirement in source,
+/// > the compiler will provide the same errors as-if they were declared explicitly in this protocol.
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:831]: `it was`

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 5]
```diff
@@ -801,15 +892,51 @@ public protocol DistributedTargetInvocationDecoder {
...
+///
+/// An instance conforming to this type must be passed when invoking
+/// ``executeDistributedTarget(on:target:invocationDecoder:handler:)`` while handling an incoming distributed call.
+///
+/// The handler will then be invoked with the return value (or error) that the invoked target returned (or threw).
+///
+/// ### Protocol requirements
+/// Similar to the ``DistributedActorSystem`` and its `remoteCall` and `remoteCallVoid` protocol requirements,
+/// the `DistributedTargetInvocationResultHandler` contains a method which is not possible to express in source due to
+/// advanced use generics combined with associated types. Specifically, the `onReturn` method is not expressed in
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:905]: `use of`

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 6]
```diff
@@ -801,15 +892,51 @@ public protocol DistributedTargetInvocationDecoder {
...
+/// source as protocol requirement, but will be treated by the compiler as-if they were.
+///
+/// > Note: Although the `onReturn` method is not expressed as protocol requirement in source,
+/// > the compiler will provide the same errors as-if they were declared explicitly in this protocol.
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:909]: `it was`

File: stdlib/public/Distributed/DistributedActorSystem.swift [hunk 7]
```diff
@@ -801,15 +892,51 @@ public protocol DistributedTargetInvocationDecoder {
...
+/// func onReturn<Success: SerializationRequirement>(value: Success) async throws
+/// ```
 @available(SwiftStdlib 5.7, *)
 public protocol DistributedTargetInvocationResultHandler {
+  /// The serialization requirement that the value passed to `onReturn` is required to conform to.
   associatedtype SerializationRequirement
+
+//  /// Ad-hoc protocol requirement
+//  ///
+//  /// Invoked when the distributed target execution returned successfully.
```
Comment (xedin) [inline stdlib/public/Distributed/DistributedActorSystem.swift:928]: `returns`?

**Comment by ktoso** [comment]
Thanks for review @xedin !

**Comment by ktoso** [comment]
Applied fixups

**Comment by ktoso** [comment]
@swift-ci please smoke test and merge

**Comment by ktoso** [comment]
@swift-ci please smoke test macOS

**Pull request merged**